dfget/core/regist/register.go

```diff
@@ -120,29 +126,15 @@ func (s *supernodeRegister) checkResponse(resp *types.RegisterResponse, e error)
 	return nil
 }
 
-func (s *supernodeRegister) setLastRegisteredNode(idx int) {
-	nLen := len(s.cfg.Nodes)
-	if nLen <= 0 {
-		return
-	}
-	if idx >= nLen {
-		s.lastRegisteredNode = ""
-		return
-	}
-	s.lastRegisteredNode = s.cfg.Nodes[idx]
+func (s *supernodeRegister) setLastRegisteredNode(node *locator.Supernode) {
+	s.lastRegisteredNode = node
 }
 
-func (s *supernodeRegister) setRemainderNodes(idx int) {
-	nLen := len(s.cfg.Nodes)
-	if nLen <= 0 {
-		return
+func (s *supernodeRegister) nodeHostStr(node *locator.Supernode) string {
```

Comment:
use `locator.Supernode.String()` function?

Reply:
the parameter node maybe nil, it will panic

Reply:
I do not think `nodeHostStr ` is a function  of type `supernodeRegister `.

Reply:
Except this, LGTM.